torch/_dynamo/variables/functions.py

```diff
@@ -117,28 +142,44 @@ def get_globals(self):
     def bind_args(self, parent, args, kwargs):
         assert not self.is_constant
         options = VariableTracker.propagate([self])
-        wrap = functools.partial(wrap_bound_arg, options=options)
-
         tx = parent.output.root_tx
+        wrap = functools.partial(wrap_bound_arg, options=options, tx=tx)
 
         fn: types.FunctionType = self.fn
+        # TODO comment
+        defaults = fn.__defaults__ or []
+        defaults_source = AttrSource(self.source, "__defaults__")
+        defaults_item_sources = [
+            GetItemSource(defaults_source, i) for i, _ in enumerate(defaults)
+        ]
```

Comment:
You do this basically twice, once for defaults and once for kw, consider if there's a helper fn somewhere here, esp so you dont have to zip in the list comprehension below.

Reply:
Yea, i actually coded up a helper but it was not convenient since i wanted to output a list for defaults and a dict for kwdefaults.  Now that I switched to a specialized DefaultsSource, it encapsulates some of the stuff and this code seems good to me without an additional helper.